src/Grids/convenience_constructors.jl

```diff
@@ -0,0 +1,366 @@
+    y_min::Real,
+    y_max::Real,
+    z_min::Real,
+    z_max::Real,
+    x1periodic::Bool,
+    x2periodic::Bool,
+    Nq::Integer,
+    n1::Integer,
+    n2::Integer,
+    device::ClimaComms.AbstractDevice = ClimaComms.device(),
+    context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+    stretch::Meshes.StretchingRule = Meshes.Uniform(),
+    hypsography::HypsographyAdaption = Flat(),
+    global_geometry::Geometry.AbstractGlobalGeometry = Geometry.CartesianGlobalGeometry(),
+    quad::Quadratures.QuadratureStyle = Quadratures.GLL{Nq}(),
+) where {FT}
+    check_device_context(context, device)
+    x1boundary = (:east, :west)
+    x2boundary = (:south, :north)
+    z_boundary_names = (:bottom, :top)
+    domain = Domains.RectangleDomain(
+        Domains.IntervalDomain(
+            Geometry.XPoint{FT}(x_min),
+            Geometry.XPoint{FT}(x_max);
+            periodic = x1periodic,
+            boundary_names = x1boundary,
+        ),
+        Domains.IntervalDomain(
+            Geometry.YPoint{FT}(y_min),
+            Geometry.YPoint{FT}(y_max);
+            periodic = x2periodic,
+            boundary_names = x2boundary,
+        ),
+    )
+    h_mesh = Meshes.RectilinearMesh(domain, n1, n2)
+    h_topology = Topologies.Topology2D(context, h_mesh)
+    h_grid = Grids.SpectralElementGrid2D(h_topology, quad)
+    z_domain = Domains.IntervalDomain(
+        Geometry.ZPoint{FT}(z_min),
+        Geometry.ZPoint{FT}(z_max);
+        boundary_names = z_boundary_names,
+    )
+    z_mesh = Meshes.IntervalMesh(z_domain, stretch; nelems = z_elem)
+    z_topology = Topologies.IntervalTopology(context, z_mesh)
+    vertical_grid = FiniteDifferenceGrid(z_topology)
+    return ExtrudedFiniteDifferenceGrid(
+        h_grid,
+        vertical_grid,
+        hypsography,
+        global_geometry,
+    )
+end
+
+"""
+    SliceXZGrid(
+        ::Type{<:AbstractFloat}; # defaults to Float64
+        z_elem::Integer,
+        x_min::Real,
+        x_max::Real,
+        z_min::Real,
+        z_max::Real,
+        x1periodic::Bool,
+        Nq::Integer,
+        n1::Integer,
+        device::ClimaComms.AbstractDevice = ClimaComms.device(),
+        context::ClimaComms.AbstractCommsContext = ClimaComms.context(device),
+        stretch::Meshes.StretchingRule = Meshes.Uniform(),
+        hypsography::HypsographyAdaption = Flat(),
+        global_geometry::Geometry.AbstractGlobalGeometry = Geometry.CartesianGlobalGeometry(),
+        quad::Quadratures.QuadratureStyle = Quadratures.GLL{Nq}(),
+    )
+
+A convenience constructor, which builds a
+`ExtrudedFiniteDifferenceGrid` with a
+`FiniteDifferenceGrid` vertical grid and a
+`SpectralElementGrid1D` horizontal grid.
+ - ``
+"""
+SliceXZGrid(; kwargs...) = SliceXZGrid(Float64; kwargs...)
+function SliceXZGrid(
+    ::Type{FT};
+    z_elem::Integer,
+    x_min::Real,
+    x_max::Real,
+    z_min::Real,
+    z_max::Real,
+    x1periodic::Bool,
+    Nq::Integer,
+    n1::Integer,
```

Comment:
```suggestion
    x_elem::Integer,
```
maybe? (For consistency)

Reply:
Done!